tfhe/src/core_crypto/gpu/slice.rs

```diff
@@ -0,0 +1,194 @@
+use crate::core_crypto::gpu::vec::range_bounds_to_start_end;
+use crate::core_crypto::gpu::{CudaPtr, CudaStream};
+use crate::core_crypto::prelude::Numeric;
+use std::ffi::c_void;
+use std::marker::PhantomData;
+use tfhe_cuda_backend::cuda_bind::{cuda_memcpy_async_gpu_to_gpu, cuda_memcpy_async_to_cpu};
+
+#[derive(Debug)]
+pub struct CudaSlice<'a, T: Numeric> {
```

Comment:
This one should also derive Copy and Clone (if the CudaPtr does not derive Copy and Clone we should add it